openc3/lib/openc3/models/interface_model.rb

```diff
@@ -281,37 +295,59 @@ def undeploy
       status_model.destroy if status_model
     end
 
-    def unmap_target(target_name)
+    def unmap_target(target_name, cmd_only: false, tlm_only: false)
+      if cmd_only and tlm_only
+        cmd_only = false
+        tlm_only = false
+      end
       target_name = target_name.to_s.upcase
 
       # Remove from this interface
-      @target_names.delete(target_name)
+      if cmd_only
+        @cmd_target_names.delete(target_name)
+        @target_names.delete(target_name) unless @tlm_target_names.include?(target_name)
+      elsif tlm_only
+        @tlm_target_names.delete(target_name)
+        @target_names.delete(target_name) unless @cmd_target_names.include?(target_name)
+      else
+        @cmd_target_names.delete(target_name)
+        @tlm_target_names.delete(target_name)
+        @target_names.delete(target_name)
+      end
       update()
 
       # Respawn the microservice
       type = self.class._get_type
       microservice_name = "#{@scope}__#{type}__#{@name}"
       microservice = MicroserviceModel.get_model(name: microservice_name, scope: scope)
-      microservice.target_names.delete(target_name)
+      microservice.target_names.delete(target_name) unless @target_names.include?(target_name)
       microservice.update
     end
 
-    def map_target(target_name)
+    def map_target(target_name, cmd_only: false, tlm_only: false, unmap_old: true)
+      if cmd_only and tlm_only
+        cmd_only = false
+        tlm_only = false
+      end
```

Comment:
How could you ever have cmd_only AND tlm_only? That doesn't logically make sense.

Reply:
Nothing keeping a user from setting both cmd_only and tlm_only so I handled that case.